app/uk/gov/hmrc/economiccrimelevyregistration/controllers/DoYouHaveCtUtrController.scala

```diff
@@ -0,0 +1,90 @@
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.economiccrimelevyregistration.connectors.EclRegistrationConnector
+import uk.gov.hmrc.economiccrimelevyregistration.controllers.actions.{AuthorisedActionWithEnrolmentCheck, DataRetrievalAction, PublicBetaAction}
+import uk.gov.hmrc.economiccrimelevyregistration.forms.DoYouHaveCtUtrFormProvider
+import uk.gov.hmrc.economiccrimelevyregistration.forms.FormImplicits.FormOps
+import uk.gov.hmrc.economiccrimelevyregistration.models.{CheckMode, Mode, Registration}
+import uk.gov.hmrc.economiccrimelevyregistration.navigation.DoYouHaveCtUtrPageNavigator
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import uk.gov.hmrc.economiccrimelevyregistration.views.html.DoYouHaveCtUtrView
+
+import javax.inject.Singleton
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class DoYouHaveCtUtrController @Inject() (
+  val controllerComponents: MessagesControllerComponents,
+  authorise: AuthorisedActionWithEnrolmentCheck,
+  getRegistrationData: DataRetrievalAction,
+  formProvider: DoYouHaveCtUtrFormProvider,
+  checkIfPublicBetaIsEnabled: PublicBetaAction,
+  eclRegistrationConnector: EclRegistrationConnector,
+  pageNavigator: DoYouHaveCtUtrPageNavigator,
+  view: DoYouHaveCtUtrView
+)(implicit
+  ec: ExecutionContext
+) extends FrontendBaseController
+    with I18nSupport {
+
+  val form: Form[Boolean] = formProvider()
+
+  def onPageLoad(mode: Mode): Action[AnyContent] =
+    (checkIfPublicBetaIsEnabled andThen authorise andThen getRegistrationData) { implicit request =>
+      Ok(view(form.prepare(request.registration.otherEntityJourneyData.isCtUtrPresent), mode))
+    }
+
+  def onSubmit(mode: Mode): Action[AnyContent] =
+    (checkIfPublicBetaIsEnabled andThen authorise andThen getRegistrationData).async { implicit request =>
+      form
+        .bindFromRequest()
+        .fold(
+          formWithErrors => Future.successful(BadRequest(view(formWithErrors, mode))),
+          answer => {
+            val userChangedIsCtutrPresentAnswer = ctutrIsNotPresentAnymore(request.registration, mode, answer)
+            val otherEntityJourneyData          =
+              request.registration.otherEntityJourneyData.copy(
+                isCtUtrPresent = Some(answer),
+                postcode = userChangedIsCtutrPresentAnswer._1,
+                ctUtr = userChangedIsCtutrPresentAnswer._2
+              )
+            eclRegistrationConnector
+              .upsertRegistration(
+                request.registration.copy(optOtherEntityJourneyData = Some(otherEntityJourneyData))
+              )
+              .map(updatedRegistration => Redirect(pageNavigator.nextPage(mode, updatedRegistration)))
+          }
+        )
+    }
+
+  private def ctutrIsNotPresentAnymore(
+    registration: Registration,
+    mode: Mode,
+    answer: Boolean
+  ): (Option[String], Option[String]) =
+    if (!answer & mode == CheckMode) {
```

Comment:
I thing you should clear `postcode` and `ctutr` if the answer is no regardless of the mode.